repondeur/zam_repondeur/tasks/periodic.py

```diff
@@ -23,4 +23,5 @@ def update_data() -> None:
 def fetch_all_amendements() -> None:
     with transaction.manager:
         for lecture in DBSession.query(Lecture):
-            fetch_amendements(lecture.pk)
+            delay = lecture.pk % (15 * 60)  # spread out updates over 15 minutes
```

Comment:
I'm not sure we should rely on the lecture's pk, maybe safer to enumerate on lectures and compute the delay based on an truly incremental index?

Reply:
The idea was that using the `lecture_pk` it allows each lecture to be updated every hour, with a stable offset. What do you think ?

Reply:
I guess I'm tired 🙂